README.md

```diff
@@ -17,34 +11,85 @@ faster, maybe. It supports specifying glue code in packages which will
 load automatically when a another package is loaded, so that explicit
 dependencies (and long load times) can be avoided.
 
-Usage is as simple as
+Suppose you've written a package called `MyPkg`. `MyPkg` has core functionality that it always provides;
+but suppose you want to provide additional functionality if the `Gadfly` package is also loaded.
+Requires.jl exports a macro, `@require`, that allows you to specify that some code is conditional on having both packages available.
+
+`@require` must be within the [`__init__`](https://docs.julialang.org/en/v1/manual/modules/#Module-initialization-and-precompilation-1) method for your module.
+Here's an example that will create a new method of a function called `myfunction` only when both packages are present:
+
+```julia
+module MyPkg
+
+# lots of code
+
+myfunction(::MyType) = Textual()
+
+function __init__()
+    @require Gadfly="c91e804a-d5a3-530f-b6f0-dfbca275c004" myfunction(::Gadfly.Plot) = Graphical()
+end
+
+end # module
+```
+
+The value in the string is Gadfly's UUID; this information may be obtained
+by finding the package in the registry ([JuliaRegistries](https://github.com/JuliaRegistries/General) for public packages).
+Note that the `Gadfly.Plot` type may not be available when you load `MyPkg`, but `@require`
+handles this situation without trouble.
+
+For larger amounts of code you can use `include` as the argument to the `@require` statement:
 
 ```julia
-media(::MyType) = Textual()
+function __init__()
+    @require Gadfly="c91e804a-d5a3-530f-b6f0-dfbca275c004" include("glue.jl")
+end
+```
+
+and this will trigger the loading and evaluation of `"glue.jl"` in `MyPkg` whenever Gadfly is loaded.
+You can even use
 
-@require Gadfly begin
-  media(::Gadfly.Plot) = Graphical()
+```julia
+function __init__()
+    @require Gadfly="c91e804a-d5a3-530f-b6f0-dfbca275c004" @eval using MyGluePkg
 end
 ```
 
-For larger amounts of code you can also use `@require Package include("glue.jl")`.
-The code wrapped by `@require` will execute as soon as the given package is loaded
-(which may be immediately).
+if you wish to exploit precompilation for the new code.
```

Comment:
So unless you package your glue code in a module it won't be precompiled?

Reply:
Correct.

Reply:
@timholy Does MyGluePkg need to be a separate Julia package with a separate repository?  Or is there a way for the MyGluePkg code to be stored in the same repo as MyPkg?

Reply:
There is no official support for that, but `MyPkg` could do a `push!(LOAD_PATH, @__DIR__)` and make itself its own little package repository.

Reply:
Would that be an acceptable approach for package authors to take, or is that considered bad practice?

Reply:
I don't have any kind of firm opinion on that, sorry.